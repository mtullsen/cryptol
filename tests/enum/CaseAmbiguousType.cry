f e =
  case e of
    _ -> ()

g e =
  case e : Integer of
    _ -> ()
