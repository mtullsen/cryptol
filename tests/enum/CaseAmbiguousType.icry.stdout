Loading module Cryptol
Loading module Cryptol
Loading module Main

[error] at CaseAmbiguousType.cry:2:8--2:9:
  Failed to infer the type of cased expression.
    Try giving the expression an explicit type annotation.
[error] at CaseAmbiguousType.cry:6:8--6:19:
  Type mismatch in cased expresson:
    Expected: an `enum` type
    Inferred: Integer
